addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5915,23 +5931,40 @@ registry.ImageTools = ImageHandlerOption.extend({
      *
      * @see this.selectClass for parameters
      */
-    async crop() {
+    crop() {
+        //Remove 'd-none' the crop butoon is the crop button is clicked
+        for (const el of this.$el.find('we-row')) {
+            el.classList.contains('croptool') && el.classList.remove('d-none')
+        }   
+        
         this.trigger_up('hide_overlay');
         this.trigger_up('disable_loading_effect');
         const img = this._getImg();
-        new weWidgets.ImageCropWidget(this, img, {mimetype: this._getImageMimetype(img)}).appendTo(this.$el[0].ownerDocument.body);
-
-        await new Promise(resolve => {
+        if(!this.ownerDocument.querySelector('.o_we_cropper_wrapper') && !window.document.querySelector('.o_we_crop_widget')){
+            this.CROPPER = new weWidgets.ImageCropWidget(this, img, {mimetype: this._getImageMimetype(img)})
+            this.CROPPER.appendTo(this.$el[0].ownerDocument.body);
+        }
+        new Promise(resolve => {
             this.$target.one('image_cropper_destroyed', async () => {
                 if (isGif(this._getImageMimetype(img))) {
                     img.dataset[img.dataset.shape ? 'originalMimetype' : 'mimetype'] = 'image/png';
                 }
-                await this._reapplyCurrentShape();
+                this._reapplyCurrentShape();
                 resolve();
             });
         });
         this.trigger_up('enable_loading_effect');
-    },
+    },  
+    /**
+     * A seperate function call was necessary for all the crop button options because 
+     * when some the changes is applied using the corp button, active class is applied 
+     * making the button UI seems active. If we make the below function calls from the
+     * crop button itself then those buttons will also becomes active, which was not the
+     * case with traditional floating crop buttons
+    */
+    transformimg(){
+        event && this.CROPPER.clickopt(event)
+    }  ,
```

Comment:
I thinks we can use `selectClass` parameters `(previewMode, widgetValue, params)`.